### PR TITLE
fix #133: allow single item list match, list not match (x!=(1,2,3)) a…

### LINKED ISSUE #133: How the FILTER works?
I want to filter one sensor, then multiple sensors then all sensors. This is how I tried to do it:
```
127.0.0.1:6379> FLUSHALL
OK
127.0.0.1:6379> TS.ADD test:1 * 1 LABELS sensor_id 1
(integer) 1560632094
127.0.0.1:6379> TS.ADD test:1 * 1 LABELS sensor_id 1
(integer) 1560632099
127.0.0.1:6379> TS.ADD test:2 * 2 LABELS sensor_id 2
(integer) 1560632129
127.0.0.1:6379> TS.ADD test:2 * 2 LABELS sensor_id 2
(integer) 1560632132
127.0.0.1:6379> TS.MRANGE 1560632041 1560632144 FILTER sensor_id=1
1) 1) "test:1"
   2) 1) 1) "sensor_id"
         2) "1"
   3) 1) 1) (integer) 1560632094
         2) "1"
      2) 1) (integer) 1560632099
         2) "1"
127.0.0.1:6379> TS.MRANGE 1560632041 1560632144 FILTER sensor_id=2
1) 1) "test:2"
   2) 1) 1) "sensor_id"
         2) "2"
   3) 1) 1) (integer) 1560632129
         2) "2"
      2) 1) (integer) 1560632132
         2) "2"
127.0.0.1:6379> TS.MRANGE 1560632041 1560632144 FILTER sensor_id!=2
(error) TSDB: please provide at least one matcher
127.0.0.1:6379> TS.MRANGE 1560632041 1560632144 FILTER sensor_id=(1,2)
(error) TSDB: please provide at least one matcher
127.0.0.1:6379> TS.MRANGE 1560632041 1560632144 FILTER sensor_id=(1)
(error) TSDB: please provide at least one matcher
127.0.0.1:6379> TS.MRANGE 1560632041 1560632144 FILTER sensor_id=(2)
(error) TSDB: please provide at least one matcher
127.0.0.1:6379> TS.MRANGE 1560632041 1560632144 FILTER sensor_id=
(error) TSDB: please provide at least one matcher
127.0.0.1:6379>
```
It looks like only `k=v` works and there is a bug or I do not understand the documentation correctly.

Thanks.

### PR DESCRIPTION
**Pull request by danni-m**
…nd empty list (same as empty label)

### CODE REVIEW

File: src/indexer.c [hunk 1]
```diff
@@ -48,35 +48,44 @@ int parsePredicate(RedisModuleCtx *ctx, RedisModuleString *label, QueryPredicate
     if (token == NULL) {
         return TSDB_ERROR;
     }
+    size_t token_len = strlen(token);
     retQuery->key = RedisModule_CreateString(ctx, token, strlen(token));
```
Comment (gkorland) [inline src/indexer.c]:
```
    retQuery->key = RedisModule_CreateString(ctx, token, token_len);
```

File: src/indexer.h [hunk 2]
```diff
@@ -21,7 +21,8 @@ typedef enum  {
     CONTAINS,
     // Not Contains a label
     NCONTAINS,
-    FILTER_LIST,  // List of predicates
+    LIST_MATCH,  // List of predicates
+    LIST_NOTMATCH,  // List of predicates
```
Comment (gkorland) [inline src/indexer.h]: LIST_NOTMATCH,  // not match a list of predicates

File: src/tests/test_module.py [hunk 3]
```diff
@@ -721,12 +721,15 @@ def test_label_index(self):
...
-            with pytest.raises(redis.ResponseError):
-                assert r.execute_command('TS.QUERYINDEX', 'generation=x', 'class=()')
+            assert r.execute_command('TS.QUERYINDEX', 'generation=x') == r.execute_command('TS.QUERYINDEX', 'generation=(x)')
+            assert r.execute_command('TS.QUERYINDEX', 'generation=x', 'class=()') == []
+            assert r.execute_command('TS.QUERYINDEX', 'class=(middle,junior,top)', 'name!=(bob,rudy,fabi)') == ['tester4']
             with pytest.raises(redis.ResponseError):
                 assert r.execute_command('TS.QUERYINDEX', 'generation=x', 'class=(')
             with pytest.raises(redis.ResponseError):
                 assert r.execute_command('TS.QUERYINDEX', 'generation=x', 'class=(ab')
+            with pytest.raises(redis.ResponseError):
```
Comment (gkorland) [inline src/tests/test_module.py]: why is that an error?

Comment (danni-m) [reply src/tests/test_module.py]: reducers only queries are not allowed

**Pull request merged**